docs/userguide/networking/index.md

```diff
@@ -1,21 +1,567 @@
+        "Scope": "local",
+        "Driver": "bridge",
+        "IPAM": {
+            "Driver": "default",
+            "Config": [
+                {}
+            ]
+        },
+        "Containers": {
+            "8c1a0a5be480921d669a073393ade66a3fc49933f08bcc5515b37b8144f6d47c": {
+                "EndpointID": "93b2db4a9b9a997beb912d28bcfc117f7b0eb924ff91d48cfa251d473e6a9b08",
+                "MacAddress": "02:42:ac:15:00:02",
+                "IPv4Address": "172.21.0.2/16",
+                "IPv6Address": ""
+            }
+        },
+        "Options": {}
+    }
+]
+```
+
+The containers you launch into this network must reside on the same Docker host.
+Each container in the network can immediately communicate with other containers
+in the network. Though, the network itself isolates the containers from external
+networks.
+
+![An isolated network](images/bridge_network.png)
+
+Within a user-defined bridge network, linking is not supported. You can
+expose and publish container ports on containers in this network. This is useful
+if you want to make a portion of the `bridge` network available to an outside
+network.
+
+![Bridge network](images/network_access.png)
+
+A bridge network is useful in cases where you want to run a relatively small
+network on a single host. You can, however, create significantly larger networks
+by creating an `overlay` network.
+
+
+### An overlay network - Docker Engine 1.12 and later
+
+As of Docker Engine 1.12, you can create an overlay network on a manager node
+running in swarm mode. You don't need an external key-value store. The swarm
+makes the overlay network available only to nodes in the swarm that require it
+for a service. When you create a service that uses the overlay network, the
+manager node automatically extends the overlay network to nodes that run service
+tasks.
+
+To learn more about running Docker Engine in swarm mode, refer to the
+[Swarm mode overview] (../../swarm/index.md).
+
+The example below shows how to create a network and use it for a service from a manager node in the swarm:
+
+```bash
+# Create an overlay network `my-multi-host-network`.
+$ docker network create --driver overlay my-multi-host-network
+
+400g6bwzd68jizzdx5pgyoe95
+
+# Create an nginx service and extend the my-multi-host-network to nodes where
+# the service's tasks run.
+$ $ docker service create --replicas 2 --network my-multi-host-network --name my-web nginx
+
+716thylsndqma81j6kkkb5aus
+```
+
+Overlay networks for a swarm are not available to containers started with
+`docker run` that don't run as part of a swarm mode service. For more
+information refer to [Note on Docker 1.12 Overlay Network Security Model](overlay-security-model.md).
+
+### An overlay network - Docker Engine 1.11 and earlier
+
+For versions of Docker Engine prior to 1.12, Docker's `overlay` network driver supports multi-host networking natively with the help of `libnetwork`, a
```

Comment:
"libnetwork" is also used for the "Swarm mode" overlay networks, so not sure we need to mention it (we also don't mention "SwarmKit" - it's an implementation detail (imo)